### ISSUE TITLE
BUG: AttributeError: type object 'dict' has no attribute 'lower'

### ISSUE THREAD

**Issue by kukushking** [modin-project/modin#6258]
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import modin.pandas as pd

df = pd.DataFrame({"a": [1, 2, 3]})

df.a.to_dict()
```

### Issue Description

In 0.22, `Series.to_dict()` throws `AttributeError: type object 'dict' has no attribute 'lower'`

### Expected Behavior

No exception, returned value `{0: 1, 1: 2, 2: 3}`

### Error Logs

<details>

```python-traceback

Traceback (most recent call last):
  File "/Users/<REDACTED>/Library/Caches/pypoetry/virtualenvs/awswrangler-NNgZqW51-py3.8/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3508, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-9-d6648240e704>", line 1, in <module>
    df.a.to_dict()
  File "/Users/<REDACTED>/Library/Caches/pypoetry/virtualenvs/awswrangler-NNgZqW51-py3.8/lib/python3.8/site-packages/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "/Users/<REDACTED>/Library/Caches/pypoetry/virtualenvs/awswrangler-NNgZqW51-py3.8/lib/python3.8/site-packages/modin/pandas/series.py", line 1922, in to_dict
    return self._query_compiler.series_to_dict(into)
  File "/Users/<REDACTED>/Library/Caches/pypoetry/virtualenvs/awswrangler-NNgZqW51-py3.8/lib/python3.8/site-packages/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "/Users/<REDACTED>/Library/Caches/pypoetry/virtualenvs/awswrangler-NNgZqW51-py3.8/lib/python3.8/site-packages/modin/core/storage_formats/base/query_compiler.py", line 495, in series_to_dict
    return self.to_pandas().to_dict(into)
  File "/Users/<REDACTED>/Library/Caches/pypoetry/virtualenvs/awswrangler-NNgZqW51-py3.8/lib/python3.8/site-packages/pandas/core/frame.py", line 1976, in to_dict
    orient = orient.lower()  # type: ignore[assignment]
AttributeError: type object 'dict' has no attribute 'lower'

```

</details>

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : eeb410c87aa657d556d3cf2403195bdc76cc7193
python           : 3.8.13.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.6.0
Version          : Darwin Kernel Version 21.6.0: Mon Dec 19 20:44:01 PST 2022; root:xnu-8020.240.18~2/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_GB.UTF-8
Modin dependencies
------------------
modin            : 0.22.1
ray              : 2.5.0
dask             : None
distributed      : None
hdk              : None
pandas dependencies
-------------------
pandas           : 1.5.3
numpy            : 1.24.3
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.8.0
pip              : 21.3.1
Cython           : None
pytest           : 7.3.2
hypothesis       : None
sphinx           : 6.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.12.2
pandas_datareader: None
bs4              : 4.12.2
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2023.6.0
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.1.2
pandas_gbq       : None
pyarrow          : 12.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : 0.4.2
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : 2023.3

</details>

**Comment by mvashishtha** [comment]
@kukushking thank you, I can reproduce the error at modin commit f617c5e6621f051d9c7befc297deec79563115db. I'll make a fix.

**Comment by kukushking** [comment]
Sounds good, thank you!

**Comment by Garra1980** [comment]
@kukushking just in case - 0.22.2 with fix for this bug and #6259 has been released
